Lesson 25

G-Memis opened the pull request.

G-Memis committed:
Extra Assignment

G-Memis committed:
i added my branch to main

G-Memis committed:
Merge branch 'main' of https://github.com/frontend-from-0/181223-exercises

G-Memis committed:
Merge branch 'main' of https://github.com/frontend-from-0/181223-exercises

G-Memis committed:
Merge branch 'main' of https://github.com/frontend-from-0/181223-exercises

G-Memis committed:
Merge branch 'main' of https://github.com/frontend-from-0/181223-exercises

G-Memis committed:
Merge branch 'main' of https://github.com/frontend-from-0/181223-exercises

G-Memis committed:
Merge branch 'main' of https://github.com/frontend-from-0/181223-exercises

G-Memis committed:
Merge branch 'main' of https://github.com/frontend-from-0/181223-exercises

G-Memis committed:
Exercises Of Lesson 25

annasvst commented on Lesson25/Gokhan/filter.js:
```diff
@@ -0,0 +1,422 @@
+	},
+];
+
+const booksPrice = booksExercise9.filter((book) => book.price < 10);
```
We want to keep books with price above 10 in this exercise.
```suggestion
const booksPrice = booksExercise9.filter((book) => book.price > 10);
```

annasvst commented on Lesson25/Gokhan/filter.js:
```diff
@@ -0,0 +1,422 @@
+	},
+];
+
+const filterUsersAge = usersExercise4.filter((users) => users.age < 25);
```
We want to keep users that has age above 25 here:
```suggestion
const filterUsersAge = usersExercise4.filter((users) => users.age > 25);
```

annasvst commented on Lesson25/Gokhan/filter.js:
```diff
@@ -0,0 +1,422 @@
+	},
+];
+
+const emptyStock = productsExercise6.filter((stock) => stock.quantity <= 0);
```
We want to keep products that are in stock, so:
```suggestion
const emptyStock = productsExercise6.filter((stock) => stock.quantity > 0);
```

annasvst commented on Lesson25/Gokhan/filter.js:
```diff
@@ -0,0 +1,422 @@
+// Given an array of numbers, filter out the numbers that are divisible by 3.
+const numbersExercise7 = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+const divisibleBy3 = numbersExercise7.filter((number) => number % 3 === 0);
```
This would also be the other way around:
```suggestion
const divisibleBy3 = numbersExercise7.filter((number) => number % 3 !== 0);
```

annasvst commented on Lesson25/Gokhan/filter.js:
```diff
@@ -0,0 +1,422 @@
+	'jungle',
+];
+
+const moreThan5Words = wordsExercise8.filter((words) => words.length <= 5);
```
```suggestion
const moreThan5Words = wordsExercise8.filter((words) => words.length > 5);
```

annasvst commented on Lesson25/Gokhan/filter.js:
```diff
@@ -0,0 +1,422 @@
+	},
+];
+
+const gradeLowerThanC = studentsExercise9.filter((students) => students.grade >= 'C');
```
You cannot compare letters reliably like this. There are different approaches to this, but one could be like this:
```suggestion
const passingGrades = ['A+', 'A', 'A-', 'B+', 'B', 'B-', 'C+', 'C'];
const gradeLowerThanC = studentsExercise9.filter((students) => !passingGrades.includes(students.grade.toUpperCase()));
```

annasvst commented on Lesson25/Gokhan/filter.js:
```diff
@@ -0,0 +1,422 @@
+	},
+];
+
+const startingNameWithJ = usersExercise10.filter((user) => user.name.startsWith ('J'));
```
Don't forget to convert the incoming value to the case that you are comparing to.
```suggestion
const startingNameWithJ = usersExercise10.filter((user) => user.name.toUpperCase().startsWith('J'));
```

annasvst reviewed (commented):
Hi Gökhan, well done overall :) I see that 'filter our' word caused some confusion. Just accept the suggestions and let me know after so I'll approve the PR.

G-Memis committed:
Update Lesson25/Gokhan/filter.js

Updated Your Commands

Co-authored-by: Anna S. Atalay <anna.atalay@minna.tech>

G-Memis committed:
Update Lesson25/Gokhan/filter.js

Updated Your Commands

Co-authored-by: Anna S. Atalay <anna.atalay@minna.tech>

G-Memis committed:
Update Lesson25/Gokhan/filter.js

Updated Your Commands

Co-authored-by: Anna S. Atalay <anna.atalay@minna.tech>

G-Memis committed:
Update Lesson25/Gokhan/filter.js

Updated Your Commands

Co-authored-by: Anna S. Atalay <anna.atalay@minna.tech>

G-Memis committed:
Update Lesson25/Gokhan/filter.js

Updated Your Commands

Co-authored-by: Anna S. Atalay <anna.atalay@minna.tech>

G-Memis committed:
Update Lesson25/Gokhan/filter.js

The result is same with your command or with my code. I did'n understand why cannot compare all with mine but changed it :)

Co-authored-by: Anna S. Atalay <anna.atalay@minna.tech>

G-Memis committed:
Updated All Last Commands

G-Memis committed:
Updeted All Last Commands

G-Memis committed:
Updated All Last Commands

annasvst reviewed (approved):
Approved ✅
